### ISSUE TITLE
Dlopen hook causes app crash on Android 12

### ISSUE THREAD

**Issue by dvgmdvgm** [vfsfitvnm/frida-il2cpp-bridge#66]
Hello.
How i can get some logs or something else cause my Android 12 smarthone just force close game after spawn.

Android 12:
Use frida spawn command `frida -U -f com.example.app -l _script.js --no-pause` just spawn my Game and cause Crashing after a second. Even cannot finish dumping command.

Use frida attach `frida -U -n Appname -l _script.js --no-pause`or `frida -U Appname  -l _script.js` works great but i cannot intercept some initial functions of the Game because should to waint until attach to the procces. dump command works great by attaching.
Same thing with frida-gadget... my default frida JS script works good.. `frida-il2cpp-bridge` script still crashed the game.

Most stranger thing that script works good on a first second and replace some function without any issue but after one second game crashes without any log/error in command prompt

Android 11 & Android 10:
`frida-Il2cpp-bridge` my script works good with frida-server and frida-gadget

Maybe i can try to update to the latest frida version for your module but idk how

**Comment by vfsfitvnm** [comment]
@dvgmdvgm it looks like the app has some anti-frida code (can it be?); did you try the same application on android 10/11?

**Comment by dvgmdvgm** [comment]
@vfsfitvnm it's all happened for the same application with the same script..
I can record some video for more clarifying//
The same script with the same Game (and totally same game version) cannot spawn the game but can be attached to the currently opened process. Problem can be reproduced only with `frida-il2cpp-bridge` script (Even just dumping command).. All of my other default frida JS script works good.
If you will try to fix this issue and you need some tests i will help with testing on my A12 and A11 smartphones

**Comment by vfsfitvnm** [comment]
@dvgmdvgm What's the game the issue is occurring on? I can try to reproduce the issue if it doesn't require google play services

**Comment by dvgmdvgm** [comment]
@vfsfitvnm _seen_

**Comment by vfsfitvnm** [comment]
@dvgmdvgm Oh well I just realized it won't work on x86 emulator, so I cannot install it. This means I cannot investigate, however you can try by yourself, you can probably start [here](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/base.ts#L59).

Does it crash with the following minimal snippet?
```ts
import "frida-il2cpp-bridge";

Il2Cpp.perform(() => {
    // this is empty
});
```

**Comment by dvgmdvgm** [comment]
@vfsfitvnm Still crash with this empty minimal snippet.
Add some console.log("Try"); after lib was founded.
It's show log in output and crases immediately..

As i say earlier, my script starts working good and can find, replace and log some initial il2cpp functions
but only initial 2, 3 functions.
It's only for A12 problem

**Comment by vfsfitvnm** [comment]
@dvgmdvgm 

 Would you edit your local copy to:

1) Replace [these two lines](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/base.ts#L80-L81) with:
```ts
setImmediate(() => {
    interceptor.detach();
    resolve();
});
```
and see what happens.

2) Comment out [this](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/base.ts#L76-L85) whole section?

**Comment by dvgmdvgm** [comment]
1. Replacing two lines give same result.

2. Commenting gives this output
```
Error: abort was called
    at attach (node_modules/frida-il2cpp-bridge/dist/il2cpp/structs/domain.js:47)
    at <anonymous> (node_modules/frida-il2cpp-bridge/dist/il2cpp/base.js:94)
Process terminated
```
And game still crashed

**Comment by vfsfitvnm** [comment]
Just to make sure, would you comment out [this](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/base.ts#L76-L85) and [this](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/base.ts#L92-L111)? And see if it crashes with the minimal snippet

**Comment by dvgmdvgm** [comment]
@vfsfitvnm yes..
Comment this two parts and game still crashes with minimal snippet without any outputs

**Comment by vfsfitvnm** [comment]
@dvgmdvgm Hmm so I guess the problem is [here](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/utils/native-wait.ts). This is good because it's not related to that specific game (or to Il2Cpp), any other app would yield this result.
Just to make sure, would you comment out also [these](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/base.ts#L72-L73) two lines (other than the two blocks of the previous comment)

**Comment by dvgmdvgm** [comment]
@vfsfitvnm yap.. Game just spawn without any issue..
But i try to execute my scrip and nothing happened. Just vanilla game was spawned

**Comment by vfsfitvnm** [comment]
@dvgmdvgm Hmm unfortunately I get this [error](https://github.com/frida/frida-java-bridge/issues/215), so I cannot use frida on my A12 emulator :/

**Comment by dvgmdvgm** [comment]
@vfsfitvnm i've got so many frida errors on my A12 until install manual edited Magisk-Frida module.
service.sh should be edited in this module to works on A12. But your emulator doesn't support Magisk root right?

**Comment by vfsfitvnm** [comment]
@dvgmdvgm Exactly, I can't install magisk

**Comment by dvgmdvgm** [comment]
@vfsfitvnm i invented some lifehack.
Just open the game and immediately lock screen, so game Activity and il2cpp was launched but initial il2cpp function not.
After that i can attach to my process with `frida-il2cpp-bridge` script, unlock screen and continue to works with il2cpp)
But still hope that's issue can be fixed natively

**Comment by vfsfitvnm** [comment]
@dvgmdvgm thanks for the update, however I can't fix this until Frida works properly on A12 emulator

**Comment by dvgmdvgm** [comment]
@vfsfitvnm which one emulator are you using?

**Comment by dvgmdvgm** [comment]
@vfsfitvnm Just get this output RN with spawn command.. Game stop Crashing and load perfectly but script stop works with following output. Attaching still works great.
```
Spawned `com.example.app`. Resuming main thread!
Error: unable to find module 'libil2cpp.so'
    at value (frida/runtime/core.js:336)
    at get module (node_modules/frida-il2cpp-bridge/dist/il2cpp/base.js:41)
    at call (native)
    at <anonymous> (node_modules/decorator-cache-getter/dist/index.js:9)
    at r (node_modules/frida-il2cpp-bridge/dist/il2cpp/api.js:1064)
    at get _domainGetAssemblies (node_modules/frida-il2cpp-bridge/dist/il2cpp/api.js:159)
    at call (native)
    at <anonymous> (node_modules/decorator-cache-getter/dist/index.js:9)
    at get assemblies (node_modules/frida-il2cpp-bridge/dist/il2cpp/structs/domain.js:17)
    at call (native)
    at <anonymous> (node_modules/decorator-cache-getter/dist/index.js:9)
    at <anonymous> (agent/index.ts:1)
    at call (native)
    at o (node_modules/browser-pack/_prelude.js:1)
    at r (node_modules/browser-pack/_prelude.js:1)
    at <eval> (/_agent.js:6196)
[Pixel 4::com.example.app]-> Process terminated
```

**Comment by vfsfitvnm** [comment]
@dvgmdvgm I'm using qemu (the one that comes with android studio), i had used android x86 and genymotion in the past, I could try them...

**Comment by dvgmdvgm** [comment]
@vfsfitvnm Gotcha.. How bout output? Gave any hints?

**Comment by vfsfitvnm** [comment]
Not very helpful at all - don't worry about that.
Just downloaded genymotion, it doesn't provide android 12 images

**Comment by brianabdl** [comment]
i try `frida-il2cpp-bridge` on A12 yesterday and it was working fine.
Maybe the issue is on your device?

**Comment by vfsfitvnm** [comment]
@Rewtio 
Uh, did you spawn the app, or just attached to it?
Also, what's your A12 ROM?

**Comment by dvgmdvgm** [comment]
@Rewtio and what's device?
I was trying on three deives

**Comment by brianabdl** [comment]
> @Rewtio Uh, did you spawn the app, or just attached to it? Also, what's your A12 ROM?

Just attach into it

**Comment by vfsfitvnm** [comment]
@dvgmdvgm fixed in 0.7.5
